modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecDialog.kt

```diff
@@ -27,59 +30,130 @@ object MirrordExecDialog {
      *
      * @return a target selected from the given list, targetlessTargetName constant if user selected targetless, null if the user cancelled
      */
-    fun selectTargetDialog(podTargets: List<String>): String? {
-        val targets = podTargets.sorted().toMutableList()
-        MirrordSettingsState.instance.mirrordState.lastChosenTarget?.let {
-            val idx = targets.indexOf(it)
-            if (idx != -1) {
-                targets.removeAt(idx)
-                targets.add(0, it)
+    fun selectTargetDialog(availableTargets: List<String>): String? {
+        val targetsState = object {
+            private var _targets = availableTargets
+                    .sorted()
+                    .toMutableList()
+                    .apply {
+                        // Initially, the last used target is at the top
+                        MirrordSettingsState.instance.mirrordState.lastChosenTarget?.let {
+                            val idx = this.indexOf(it)
+                            if (idx != -1) {
+                                this.removeAt(idx)
+                                this.add(0, it)
+                            }
+                        }
+                    }
+                    .toList()
+            val targets: List<String>
+                get() {
+                    return this._targets + targetlessTargetName
+                }
+            var pods = true
+                set(value) {
+                    field = value
+                    this.update()
+                }
+            var deployments = true
+                set(value) {
```

Comment:
Are these persistent? Does it remember what categories I had checked the next time I run mirrord?

Reply:
Now they are :)